modules/iam-assumable-role-with-oidc/main.tf

```diff
@@ -62,6 +62,28 @@ data "aws_iam_policy_document" "assume_role_with_oidc" {
   }
 }
 
+data "aws_iam_policy_document" "assume_self" {
+  count = var.explicit_permission_to_assume_self ? 1 : 0
+
+  statement {
+    actions = [
+      "sts:AssumeRole",
+    ]
+
+    principals {
+      type        = "AWS"
+      identifiers = ["arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/${var.role_name}"]
```

Comment:
note - be careful because role path is not clear here and unless users know to pre-pend the role path into `var.role_name`, they may face issues when using roles with role paths

Reply:
Maybe we should include `var.role_path` also into this line?